pkg/ddc/base/runtime_helper.go

```diff
@@ -67,30 +74,62 @@ func (info *RuntimeInfo) GetTemplateToInjectForFuse(pvcName string, enableCacheD
 		ds.Spec.Template.Spec.Volumes = utils.TrimVolumes(ds.Spec.Template.Spec.Volumes, cacheDirNames)
 	}
 
-	// 1. set the pvc name
+	// 1. setup fuse sidecar container when enabling virtual fuse device
+	if enableVirtFuseDev {
+		// remove the fuse related volumes if using virtual fuse device
+		ds.Spec.Template.Spec.Containers[0].VolumeMounts = utils.TrimVolumeMounts(ds.Spec.Template.Spec.Containers[0].VolumeMounts, hostMountNames)
```

Comment:
Could you put `enableVirtFuseDev` logics into a separate function？

Reply:
Done